src/bots/jitMaker.ts

```diff
@@ -280,447 +168,413 @@ export class JitMakerBot implements Bot {
+					const driftUser = this.driftClient.getUser(subId);
+					const perpMarketAccount =
+						this.driftClient.getPerpMarketAccount(perpIdx);
+					const oraclePriceData =
+						this.driftClient.getOracleDataForPerpMarket(perpIdx);
+
+					const maxBase: number = calculateBaseAmountToMarketMake(
+						perpMarketAccount,
+						driftUser.getNetSpotMarketValue(),
+						1
+					);
 
-			// update current position based on market position
-			this.agentState.perpMarketPosition.set(p.marketIndex, p);
+					const dollarDepth = 1000; // todo
 
-			// update state
-			let currentState = this.agentState.stateType.get(p.marketIndex);
-			if (!currentState) {
-				this.agentState.stateType.set(p.marketIndex, StateType.NEUTRAL);
-				currentState = StateType.NEUTRAL;
-			}
+					// const baseDepth = new BN(dollarDepth * QUOTE_PRECISION.toNumber()).mul(BASE_PRECISION).div(oraclePriceData.price);
+					// const userPerpPos: PerpPosition = driftUser.getPerpPosition(perpIdx);
 
-			let canUpdateStateBasedOnPosition = true;
-			if (
-				(currentState === StateType.CLOSING_LONG &&
-					p.baseAssetAmount.gt(new BN(0))) ||
-				(currentState === StateType.CLOSING_SHORT &&
-					p.baseAssetAmount.lt(new BN(0)))
-			) {
-				canUpdateStateBasedOnPosition = false;
-			}
+					const perpMarketIndex = perpIdx;
 
-			if (canUpdateStateBasedOnPosition) {
-				// check if need to enter a closing state
-				const accountCollateral = convertToNumber(
-					this.driftClient.getUser().getTotalCollateral(),
-					QUOTE_PRECISION
-				);
-				const positionValue = convertToNumber(
-					p.quoteAssetAmount,
-					QUOTE_PRECISION
-				);
-				const exposure = positionValue / accountCollateral;
-
-				if (exposure >= this.MAX_POSITION_EXPOSURE) {
-					// state becomes closing only
-					if (p.baseAssetAmount.gt(new BN(0))) {
-						this.agentState.stateType.set(
-							p.marketIndex,
-							StateType.CLOSING_LONG
-						);
-					} else {
-						this.agentState.stateType.set(
-							p.marketIndex,
-							StateType.CLOSING_SHORT
-						);
+					if (perpIdx == 0) {
+						spotMarketIndex = 1;
+					} else if (perpIdx == 1) {
+						spotMarketIndex = 3;
+					} else if (perpIdx == 2) {
+						spotMarketIndex = 4;
 					}
-				} else {
-					// update state to be whatever our current position is
-					if (p.baseAssetAmount.gt(new BN(0))) {
-						this.agentState.stateType.set(p.marketIndex, StateType.LONG);
-					} else if (p.baseAssetAmount.lt(new BN(0))) {
-						this.agentState.stateType.set(p.marketIndex, StateType.SHORT);
-					} else {
-						this.agentState.stateType.set(p.marketIndex, StateType.NEUTRAL);
-					}
-				}
-			}
-		}
-	}
 
-	private nodeCanBeFilled(
-		node: DLOBNode,
-		userAccountPublicKey: PublicKey
-	): boolean {
-		if (node.haveFilled) {
-			logger.error(
-				`already made the JIT auction for ${node.userAccount} - ${node.order.orderId}`
-			);
-			return false;
-		}
+					this.jitter.setUserFilter((userAccount, userKey) => {
```

Comment:
might be good to filter out our user account key too

Reply:
done